packages/server/main.js

```diff
@@ -70,35 +68,47 @@ const METHODS = {
   },
 };
 
-app.use(async (ctx, next) => {
-  if (ctx.path !== "/api/") {
-    return next();
-  }
+wss.on("connection", function(wss) {
+  clients.push(wss);
+  console.log("Connected");
+
+  wss.on("message", async function(req) {
+    try {
+      const request = parse(req);
+      if (request.type !== "request") {
+        throw new InvalidRequest();
+      }
+
+      const method = METHODS[request.method];
+      if (method === undefined) {
+        throw new MethodNotFound(request.method);
+      }
+
+      var result = method(request.params);
+      wss.send(
+        format.response(request.id, result === undefined ? true : result)
+      );
 
-  try {
-    const request = parse(await getStream(ctx.req));
-    if (request.type !== "request") {
-      throw new InvalidRequest();
+      if (request.method !== "listEntries") {
+        result = METHODS["listEntries"]();
+        clients.forEach(function(client) {
+          if (client !== wss) {
+            client.send(format.response(request.id, result));
```

Comment:
This should not be a response, a response must correspond to a request (that's where there is `request.id`) but these clients receive a response without having sent a request.

It should be [a notification](https://github.com/JsCommunity/json-rpc-protocol#notification).